src/symbol.rs

```diff
@@ -101,6 +101,15 @@ impl Symbol {
         }
     }
 
+    /// Converts this symbol into a shared one
+    pub(crate) fn into_shared(self) -> Symbol {
+        match self.text {
+            SymbolText::Shared(text) => Symbol::shared(text),
+            SymbolText::Owned(text) => Symbol::shared(text.into()),
+            SymbolText::Unknown => Symbol::unknown_text(),
+        }
+    }
```

Comment:
🗺️ This is really needed for supporting `IonReader::field_name()`  The return of this method wants a `Symbol`, but when we adapt it, we don't want to have to generate a copy for repeated calls to it, so this allows us to intern it if we have to return it after materializing it once.